front-end/src/screens/ReferendumPost/query.ts

```diff
@@ -47,6 +47,7 @@ const referendumPost = gql`
         }
         content
         created_at
+        has_poll
```

Comment:
not needed
```suggestion
```

Reply:
typescript complaining if it is not there

Reply:
Can you tell me more where it's no happy? Then we do something wrong. Depending on what is called, the types should be different. We should not over-fetch.

Reply:
Property 'has_poll' does not exist on type '({ __typename?: "posts" | undefined; } & { __typename?: "posts" | undefined; } & Pick<Posts, "title" | "id" | "content" | "created_at" | "updated_at"> & { author: Maybe<...>; comments: ({ ...; } & ... 2 more ... & { ...; })[]; onchain_link: Maybe<...>; topic: { ...; } & Pick<...>; type: { ...; } & Pick<...>; }) | ({...'.
  Property 'has_poll' does not exist on type '{ __typename?: "posts" | undefined; } & { __typename?: "posts" | undefined; } & Pick<Posts, "title" | "id" | "content" | "created_at" | "updated_at"> & { author: Maybe<...>; comments: ({ ...; } & ... 2 more ... & { ...; })[]; onchain_link: Maybe<...>; topic: { ...; } & Pick<...>; type: { ...; } & Pick<...>; }'.ts(2339)

Reply:
post is data.posts[0]

data type is 

data: (
		DiscussionPostAndCommentsQueryHookResult['data'] |
		ProposalPostAndCommentsQueryHookResult['data'] |
		ReferendumPostAndCommentsQueryHookResult['data'] |
		MotionPostAndCommentsQueryHookResult['data'] |
		TreasuryProposalPostAndCommentsQueryHookResult['data']
	)

Reply:
any one of them missing any key and typescript will complaint

Reply:
Then we need to convince the compiler that we know better than him.
Either wit as `as DiscussionPostAndCommentsQueryHookResult['data'] ` or with type guard.
I've pushed a fix